src/OpenCLScheduler.h

```diff
@@ -36,22 +37,13 @@ class OpenCLScheduler {
                  std::vector<net_t>& output_pol,
                  std::vector<net_t>& output_val);
 private:
-    class ForwardTask {
-    public:
-        const std::vector<net_t> *input;
-        std::vector<net_t> * output;
-        std::promise<void> prom;
-        ForwardTask() : input(nullptr), output(nullptr) {}
-        ForwardTask(const std::vector<net_t> * in,
-                    std::vector<net_t> * out)
-            : input(in), output(out) {}
-    };
-
     std::vector<std::unique_ptr<OpenCL_Network>> m_networks;
     std::vector<std::unique_ptr<OpenCL>> m_opencl;
-    Utils::ThreadPool m_threadpool;
+    // XXX : I failed to figure out how to make this compile with unique_ptr
+    // especially on visual studio
+    std::vector<std::list<std::shared_ptr<OpenCLContext>>> m_context;
```

Comment:
I might make sense to make a helper type for the innermost list. Would also make it clearer why 2 types of collections are being nested here.